### PR TITLE
Fix reader related posts loading the wrong post after rotation

### LINKED ISSUE #14195: Reader:  Wrong related post shown on rotation
### Expected behavior

Given the user is on a Reader related post screen (nested at least to 2 levels), on rotating the device, last selected related post should be shown.

### Actual behavior

Given the user is on a "Reader" related post screen (nested at least to 2 levels), on rotating the device, first selected related post is shown instead of the last one. 

### Steps to reproduce the behavior

1. Open a wp.com post on Reader tab.
2. Scroll to bottom of the post details page.
3. Notice that related posts are displayed.
4. Click a related post. 
5. Notice that selected related post details are shown on "Related Post" screen with a cross button.
6. Open another related post from the previous "Related Post" screen.
7. Notice that selected related post details are shown on "Related Post" screen with a cross button. 
8. Rotate the device.
9. Notice that first related post is shown instead of the last one.

https://user-images.githubusercontent.com/1405144/110094941-64035700-7dc2-11eb-9931-961069cf83a9.mp4

More details in: https://github.com/wordpress-mobile/WordPress-Android/pull/14194#issuecomment-791296924

### PR DESCRIPTION
**Pull request by dnalves**
Fixes #14195 

### Issue description
There's an issue that when using the reader to see Related Posts, after going more than 2 levels deep in related posts and rotating the device, the post loaded will be the first one displayed.

### Work done
I changed where the Fragment restores its state from `onActivityCreated` to `onViewCreated`.
`onActivityCreated` is called after `onViewCreated`, so the state was being restored after the view was ready and the post was already loaded with the wrong id.

### To test
1. Open a wp.com post on Reader tab.
2. Scroll to bottom of the post details page.
3. Notice that related posts are displayed.
4. Click a related post.
5. Notice that selected related post details are shown on "Related Post" screen with a cross button.
6. Open another related post from the previous "Related Post" screen.
7. Notice that selected related post details are shown on "Related Post" screen with a cross button.
8. Rotate the device.
9. Notice that first related post is shown instead of the last one.

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None

3. What automated tests I added (or what prevented me from doing so)
None. No automated tests are present for the area (just adding them should be a separate PR)

PR submission checklist:

- [X] I have completed the Regression Notes.
- [X] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: RELEASE-NOTES.txt
Comment (jd-alexander) [inline RELEASE-NOTES.txt]: @dnalves Thanks for making an entry here! For these entries we try to write them in the past tense when possible. So something like this would suffice: "Fixed an issue where navigating through a couple of related posts in Reader caused the incorrect post to be loaded after rotation". Let me know what you think. 

Our release notes are sent to the Editorial team, so they are reviewed before being published to the Play Store.

Comment (dnalves) [reply RELEASE-NOTES.txt]: Makes a lot of sense! Thanks for the suggestion. And I'm glad those are reviewed, I wouldn't trust myself with final copies 😂 

I'll make the change :D

Comment (jd-alexander) [reply RELEASE-NOTES.txt]: Thanks @dnalves 😄

**Comment by AliSoftware** [comment]
Thanks @dnalves for submitting this PR! 🙇 

I am doing the Code Freeze for WPAndroid 18.4 today. I'm thus moving this PR's milestone to 18.5

@jd-alexander I see you reviewed the PR previously, I'll let you decide if this needs to go in 18.4 – and in that case, we will want to update the PR to target the upcoming `release/18.4` branch to get it included in 18.4, revert the milestone to 18.4, and do a new beta after the one I'll do post-code-freeze – or if you're ok moving this to 18.5 so it has more time to get reviewed and tested.

**Review by jd-alexander — COMMENTED** [review]
@dnalves this PR was bumped to be shipped in 18.5. Could you update the release notes entry to this section? Thank you. 

Otherwise from that, I tested the PR and the changes worked as expected. Good fix! I will approve and merge once the changes are made. 🚢

**Comment by dnalves** [comment]
> @dnalves this PR was bumped to be shipped in 18.5. Could you update the release notes entry to this section? Thank you.
> 
> Otherwise from that, I tested the PR and the changes worked as expected. Good fix! I will approve and merge once the changes are made. 🚢

Done 👍 
Thank you! :D

**Review by jd-alexander — APPROVED** [review]
Tested and working as expected. Thanks for this fix @dnalves LGTM 🚢

**Pull request merged**
